numerical_methods/false_position.cpp

```diff
@@ -13,62 +18,112 @@
+namespace numerical_methods {
+/**
+ * @namespace false_position
+ * @brief Functions for [False Position]
+ * (https://en.wikipedia.org/wiki/Regula_falsi) method.
+ */
+namespace false_position {
+/**
+* @brief This function gives the value of f(x) for given x.
+* @param x value for which we have to find value of f(x).
+* @return value of f(x) for given x.
+*/  
+static double eq(double x) {
+    return (x*x-x);  // original equation
 }
 
-/** main function */
-int main() {
-    double a = -1, b = 1, x, z, m, n, c;
-    int i;
-
-    // loop to find initial intervals a, b
-    for (int i = 0; i < MAX_ITERATIONS; i++) {
-        z = eq(a);
-        x = eq(b);
-        if (sgn(z) == sgn(x)) {  // same signs, increase interval
-            b++;
-            a--;
-        } else {  // if opposite signs, we got our interval
-            break;
+/**
+* @brief This function finds root of the equation in given interval i.e. (x1,x2).
+* @param x1,x2 values for an interval in which root is present.
+  @param y1,y2 values of function at x1, x2 espectively.
+* @return root of the equation in the given interval.
+*/  
+static double regula_falsi(double x1,double x2,double y1,double y2){
+    double diff = x1-x2;
+    if(diff<0){
+        diff= (-1)*diff;
+    }
+    if(diff<0.00001){          
+        if (y1<0) {
+            y1=-y1;
+        }
+        if (y2<0) {
+            y2=-y2;
+        }
+        if (y1<y2) {
+            return x1;
+        }
+        else {
+            return x2;
         }
     }
+    double x3=0,y3=0;
+    x3 = x1 - (x1-x2)*(y1)/(y1-y2);
+    y3 = eq(x3);
+    return regula_falsi(x2,x3,y2,y3);
+}
 
-    std::cout << "\nFirst initial: " << a;
-    std::cout << "\nSecond initial: " << b;
-
-    for (i = 0; i < MAX_ITERATIONS; i++) {
-        m = eq(a);
-        n = eq(b);
-
-        c = ((a * n) - (b * m)) / (n - m);
+/**
+* @brief This function prints roots of the equation.
+* @param root which we have to print. 
+* @param count which is count of the root in an interval [-range,range].
+*/  
+void printRoot(double root, const int16_t &count){
+    if(count==1){
+        std::cout << "Your 1st root is : " << root << std::endl;
+    }
+    else if(count==2){
+        std::cout << "Your 2nd root is : " << root << std::endl;
+    }
+    else if(count==3){
+        std::cout << "Your 3rd root is : " << root << std::endl;
+    }
+    else{
+        std::cout << "Your "<<count<<"th root is : " << root << std::endl;
+    }
+}
+}  // namespace false_position
+}  // namespace numerical_methods
 
-        a = c;
-        z = eq(c);
 
-        if (std::abs(z) < EPSILON) {  // stoping criteria
-            break;
+/**
+* @brief Main function
+* @returns 0 on exit
+*/
+int main() {
+    double a=0, b=0,i=0,root=0;
+    int16_t count=0;
+    int32_t range = 100000;        //Range in which we have to find the root. (-range,range)
+    double gap = 0.5;          //interval gap. lesser the gap more the accuracy
```

Comment:
`float` would be better here I think.
```suggestion
    float gap = 0.5;          // interval gap. lesser the gap more the accuracy
```